libs/application/ui-shell/src/lib/ApplicationForm.tsx

```diff
@@ -35,48 +35,70 @@ const ShellWrapper: FC<{
 
   const application = data?.getApplication
 
+  if (!applicationId || error) {
+    return <NotFound />
+  }
+  // TODO we need better loading states
+  if (loading) {
+    return null
+  }
+  return (
+    <ShellWrapper
+      application={application}
+      nationalRegistryId={nationalRegistryId}
+    />
+  )
+}
+
+const ShellWrapper: FC<{
+  application: Application
+  nationalRegistryId: string
+}> = ({ application, nationalRegistryId }) => {
+  const [dataSchema, setDataSchema] = useState<Schema>()
+  const [form, setForm] = useState<Form>()
+  const [, fieldsDispatch] = useFields()
+
   useEffect(() => {
     async function populateForm() {
-      if (application !== undefined && form === undefined) {
+      console.log('popluate')
+      if (dataSchema === undefined && form === undefined) {
+        console.log('load templates')
         const template = await getApplicationTemplateByTypeId(
           application.typeId,
         )
         if (
           template !== null &&
           !(isOnProduction() && !template.readyForProduction)
         ) {
-          const stateInformation = await getApplicationStateInformation(
-            application,
-          )
-          const applicationFields = await getApplicationUIFields(
-            application.typeId,
-          )
-          fieldsDispatch(applicationFields)
```

Comment:
I think this is the bug actually. This changes the surrounding context, triggering a re-render of the whole tree. Then, this effect is run again because `form` is still undefined because the `formLoader` is being awaited still. This might be setting of a infinity loop.

I still might be wrong, and it is just a bug in how we lazy load templates and forms.